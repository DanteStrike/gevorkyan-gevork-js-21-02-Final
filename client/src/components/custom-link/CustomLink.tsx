import React from 'react';
import './CustomLink.scss';
import {Link, LinkProps} from 'react-router-dom';

interface ICustomLink extends LinkProps {
  className?: string;
  children?: React.ReactNode;
}

function CustomLink({children = ``, className = ``, ...linkProps}: ICustomLink) {
  return (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <Link className={`custom-link ${className}`.trim()} {...linkProps}>
      {children}
    </Link>
  );
}

CustomLink.defaultProps = {
  className: ``,
  children: ``,
};

export default React.memo(CustomLink);
