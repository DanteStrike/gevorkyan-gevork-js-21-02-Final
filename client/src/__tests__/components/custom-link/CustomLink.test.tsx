import {shallow} from 'enzyme';
import React from 'react';
import CustomLink from '../../../components/custom-link/CustomLink';

describe(`Component CustomLink should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(
      <CustomLink to="/route" className="link">
        <div id="content" />
      </CustomLink>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
