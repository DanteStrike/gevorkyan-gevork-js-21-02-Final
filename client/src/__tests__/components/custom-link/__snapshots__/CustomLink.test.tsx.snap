// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CustomLink should render correctly snapshot default test 1`] = `
<Link
  className="custom-link link"
  to="/route"
>
  <div
    id="content"
  />
</Link>
`;
